---
abstract: |
  During the last decade, trajectory inference (TI) methods have emerged as a novel framework to model cell developmental dynamics, most notably in the area of single-cell transcriptomics. At present, more than 70 TI methods have been published, and recent benchmarks showed that even state-of-the-art methods only perform well for certain trajectory types but not others. In this work, we present TinGa, a new TI model that is fast and flexible, and that is based on Growing Neural Graphs. We performed an extensive comparison of TinGa to five state-of-the-art methods for TI on a set of 250 datasets, including both synthetic as well as real datasets. Overall, TinGa improves the state-of-the-art by producing accurate models (comparable to or an improvement on the state-of-the-art) on the whole spectrum of data complexity, from the simplest linear datasets to the most complex disconnected graphs. In addition, TinGa obtained the fastest execution times, showing that our method is thus one of the most versatile methods up to date.
authors:
- Helena Todorov
- rcannood
- Wouter Saelens
- Yvan Saeys
date: "2020-07-01T00:00:00Z"
doi: "10.1093/bioinformatics/btaa463"
featured: false
image:
  caption: 'Trajectories found by the different methods on a real dataset with a linear trajectory.'
  focal_point: ''
  preview_only: false
project: []
publication: 'Bioinformatics'
publication_short: ''
publication_types:
- '2'
publishDate: '2020-07-01T00:00:00Z'
# slides: example
summary: TinGa, a new TI model that is fast and flexible, and that is based on Growing Neural Graphs.
# tags:
# - Example Tag
title: 'TinGa: fast and flexible trajectory inference with Growing Neural Gas'
url_code: 'https://github.com/Helena-todd/TinGa'
# url_dataset: ''
# url_pdf: ''
# url_poster: ''
# url_slides: ''
# url_source: ''
# url_video: ''
---
