---
abstract: 'When developing new types of tools for single-cell analyses, there is often a lack of datasets on which to quantitatively assess the performance. We developed dyngen, a multi-modality simulator of single cells. In dyngen, the biomolecular state of an in silico changes over time according to a predefined gene regulatory network. We used dyngen to benchmark three emerging ways of analysing single-cell data: RNA velocity, cell-specific network inference and trajectory alignment methods. dyngen lays the foundations for benchmarking a wide variety of computational single-cell tools and can be used to help kick-start the development of future types of analyses.'
authors:
- rcannood
- Wouter Saelens
- Louise Deconinck
- Yvan Saeys
author_notes:
- Equal contribution
- Equal contribution
date: "2020-02-07T00:00:00Z"
doi: 10.1101/2020.02.06.936971
featured: false
image:
  caption: 'Showcase of dyngen functionality.'
  focal_point: ''
  preview_only: false
project: []
publication: 'bioRxiv'
publication_short: ''
publication_types:
- '3'
publishDate: '2020-02-07T00:00:00Z'
# slides: example
summary: Simulate gene regulation, splicing and translation at a single cell, single molecule level.
# tags:
# - Example Tag
title: 'dyngen: a multi-modal simulator for spearheading new single-cell omics analyses'
url_code: 'https://www.github.com/dynverse/dyngen'
# url_dataset: ''
url_pdf: 'https://raw.githubusercontent.com/dynverse/dyngen_manuscript/master/manuscript/manuscript.pdf'
# url_poster: ''
# url_slides: ''
# url_source: ''
# url_video: ''
---
