---
title: "anndata for R"
author: rcannood
date: 2021-02-02T20:53:00+00:00
categories: ["Data science"]
tags: ["Data science", "Bioinformatics", "scanpy", "Transcriptomics", "R"]
summary: "Porting anndata to R with reticulate."
subtitle: "Porting anndata to R with reticulate."
image:
  caption: "anndata provides a scalable way of keeping track of data and learned annotations."
bibliography: references.bib
---

<script src="index_files/header-attrs/header-attrs.js"></script>


<p><a href="https://anndata.readthedocs.io/en/latest/"><code>anndata</code></a> is a commonly used Python package
for keeping track of data and learned annotations, and can be used to read from and write to the h5ad
file format. It is also the main data format used in the scanpy python package <span class="citation">(Wolf, Angerer, and Theis 2018)</span>.</p>
<p>However, using scanpy/anndata in R can be a major hassle. When trying to read an h5ad file,
R users could approach this problem in one of two ways. A) You could read in the file manually
(since it’s an H5 file), but this involves a lot of manual work and a lot of understanding
on how the h5ad and H5 file formats work (also, expect major headaches from cryptic hdf5r bugs).
Or B) interact with scanpy and anndata through reticulate, but run into issues converting
some of the python objects into R.</p>
<p>We recently published <a href="https://cran.r-project.org/package=anndata"><code>anndata</code></a> on CRAN,
which is an R wrapper for the Python package – with some syntax sprinkled on top to make
R users feel more at home.</p>
<p>The R anndata is still under active development at <a href="https://github.com/dynverse/anndata">github.com/dynverse/anndata</a>.
If you encounter any issues, feel free to post an issue on GitHub!</p>
<div id="installation" class="section level2">
<h2>Installation</h2>
<p>Installing <code>anndata</code> is not particularly hard. You do need R and Python to be installed.
If you don’t have a working installation of Python, you can let reticulate install Miniconda.</p>
<pre class="r"><code># install the R anndata package
install.packages(&quot;anndata&quot;)

# skip this if you wish to use the local python installation
reticulate::install_miniconda()

# skip this if anndata is already installed
anndata::install_anndata()</code></pre>
</div>
<div id="getting-started" class="section level2">
<h2>Getting started</h2>
<p>The API of <code>anndata</code> is very similar to its Python counterpart.
Check out <code>?anndata</code> for a full list of the functions provided by this package.</p>
<p><code>AnnData</code> stores a data matrix <code>X</code> together with annotations
of observations <code>obs</code> (<code>obsm</code>, <code>obsp</code>), variables <code>var</code> (<code>varm</code>, <code>varp</code>),
and unstructured annotations <code>uns</code>.</p>
<p>Here is an example of how to create an AnnData object with 2 observations and 3 variables.</p>
<pre class="r"><code>library(anndata)

ad &lt;- AnnData(
  X = matrix(1:6, nrow = 2),
  obs = data.frame(group = c(&quot;a&quot;, &quot;b&quot;), row.names = c(&quot;s1&quot;, &quot;s2&quot;)),
  var = data.frame(type = c(1L, 2L, 3L), row.names = c(&quot;var1&quot;, &quot;var2&quot;, &quot;var3&quot;)),
  layers = list(
    spliced = matrix(4:9, nrow = 2),
    unspliced = matrix(8:13, nrow = 2)
  ),
  obsm = list(
    ones = matrix(rep(1L, 10), nrow = 2),
    rand = matrix(rnorm(6), nrow = 2),
    zeros = matrix(rep(0L, 10), nrow = 2)
  ),
  varm = list(
    ones = matrix(rep(1L, 12), nrow = 3),
    rand = matrix(rnorm(6), nrow = 3),
    zeros = matrix(rep(0L, 12), nrow = 3)
  ),
  uns = list(
    a = 1, 
    b = data.frame(i = 1:3, j = 4:6, value = runif(3)),
    c = list(c.a = 3, c.b = 4)
  )
)

ad</code></pre>
<pre><code>## AnnData object with n_obs × n_vars = 2 × 3
##     obs: &#39;group&#39;
##     var: &#39;type&#39;
##     uns: &#39;a&#39;, &#39;b&#39;, &#39;c&#39;
##     obsm: &#39;ones&#39;, &#39;rand&#39;, &#39;zeros&#39;
##     varm: &#39;ones&#39;, &#39;rand&#39;, &#39;zeros&#39;
##     layers: &#39;spliced&#39;, &#39;unspliced&#39;</code></pre>
<p>You can read the information back out using the <code>$</code> notation.</p>
<pre class="r"><code>ad$X</code></pre>
<pre><code>##    var1 var2 var3
## s1    1    3    5
## s2    2    4    6</code></pre>
<pre class="r"><code>ad$obs</code></pre>
<pre><code>##    group
## s1     a
## s2     b</code></pre>
<pre class="r"><code>ad$obsm[[&quot;ones&quot;]]</code></pre>
<pre><code>##      [,1] [,2] [,3] [,4] [,5]
## [1,]    1    1    1    1    1
## [2,]    1    1    1    1    1</code></pre>
<pre class="r"><code>ad$layers[[&quot;spliced&quot;]]</code></pre>
<pre><code>##    var1 var2 var3
## s1    4    6    8
## s2    5    7    9</code></pre>
<pre class="r"><code>ad$uns[[&quot;b&quot;]]</code></pre>
<pre><code>##   i j     value
## 1 1 4 0.1316952
## 2 2 5 0.9088103
## 3 3 6 0.6214928</code></pre>
<div id="using-scanpy" class="section level3">
<h3>Using scanpy</h3>
<p>A nice side-effect of loading this package is that it is now super easy to
interact with scanpy through these AnnData objects.</p>
<pre class="r"><code>library(reticulate)

ad$X</code></pre>
<pre><code>##    var1 var2 var3
## s1    1    3    5
## s2    2    4    6</code></pre>
<pre class="r"><code>sc &lt;- import(&quot;scanpy&quot;)
sc$pp$normalize_per_cell(ad)

ad$X</code></pre>
<pre><code>##        var1 var2     var3
## s1 1.166667  3.5 5.833333
## s2 1.750000  3.5 5.250000</code></pre>
</div>
<div id="reading-writing-anndata-objects" class="section level3">
<h3>Reading / writing AnnData objects</h3>
<p>Read from h5ad format:</p>
<pre class="r"><code>read_h5ad(&quot;pbmc_1k_protein_v3_processed.h5ad&quot;)</code></pre>
<pre><code>## AnnData object with n_obs × n_vars = 713 × 33538
##     var: &#39;gene_ids&#39;, &#39;feature_types&#39;, &#39;genome&#39;, &#39;highly_variable&#39;, &#39;means&#39;, &#39;dispersions&#39;, &#39;dispersions_norm&#39;
##     uns: &#39;hvgParameters&#39;, &#39;normalizationParameters&#39;, &#39;pca&#39;, &#39;pcaParameters&#39;
##     obsm: &#39;X_pca&#39;
##     varm: &#39;PCs&#39;</code></pre>
</div>
<div id="creating-a-view" class="section level3">
<h3>Creating a view</h3>
<p>You can use any of the regular R indexing methods to subset the <code>AnnData</code> object.
This will result in a ‘View’ of the underlying data without needing to store
the same data twice.</p>
<pre class="r"><code>view &lt;- ad[, 2]
view</code></pre>
<pre><code>## View of AnnData object with n_obs × n_vars = 2 × 1
##     obs: &#39;group&#39;, &#39;n_counts&#39;
##     var: &#39;type&#39;
##     uns: &#39;a&#39;, &#39;b&#39;, &#39;c&#39;
##     obsm: &#39;ones&#39;, &#39;rand&#39;, &#39;zeros&#39;
##     varm: &#39;ones&#39;, &#39;rand&#39;, &#39;zeros&#39;
##     layers: &#39;spliced&#39;, &#39;unspliced&#39;</code></pre>
<pre class="r"><code>view$is_view</code></pre>
<pre><code>## [1] TRUE</code></pre>
<pre class="r"><code>ad[,c(&quot;var1&quot;, &quot;var2&quot;)]</code></pre>
<pre><code>## View of AnnData object with n_obs × n_vars = 2 × 2
##     obs: &#39;group&#39;, &#39;n_counts&#39;
##     var: &#39;type&#39;
##     uns: &#39;a&#39;, &#39;b&#39;, &#39;c&#39;
##     obsm: &#39;ones&#39;, &#39;rand&#39;, &#39;zeros&#39;
##     varm: &#39;ones&#39;, &#39;rand&#39;, &#39;zeros&#39;
##     layers: &#39;spliced&#39;, &#39;unspliced&#39;</code></pre>
<pre class="r"><code>ad[-1, ]</code></pre>
<pre><code>## View of AnnData object with n_obs × n_vars = 1 × 3
##     obs: &#39;group&#39;, &#39;n_counts&#39;
##     var: &#39;type&#39;
##     uns: &#39;a&#39;, &#39;b&#39;, &#39;c&#39;
##     obsm: &#39;ones&#39;, &#39;rand&#39;, &#39;zeros&#39;
##     varm: &#39;ones&#39;, &#39;rand&#39;, &#39;zeros&#39;
##     layers: &#39;spliced&#39;, &#39;unspliced&#39;</code></pre>
</div>
<div id="anndata-as-a-matrix" class="section level3">
<h3>AnnData as a matrix</h3>
<p>The <code>X</code> attribute can be used as an R matrix:</p>
<pre class="r"><code>ad$X[,c(&quot;var1&quot;, &quot;var2&quot;)]</code></pre>
<pre><code>##        var1 var2
## s1 1.166667  3.5
## s2 1.750000  3.5</code></pre>
<pre class="r"><code>ad$X[-1, , drop = FALSE]</code></pre>
<pre><code>##    var1 var2 var3
## s2 1.75  3.5 5.25</code></pre>
<pre class="r"><code>ad$X[, 2] &lt;- 10</code></pre>
<p>You can access a different layer matrix as follows:</p>
<pre class="r"><code>ad$layers[&quot;unspliced&quot;]</code></pre>
<pre><code>##    var1 var2 var3
## s1    8   10   12
## s2    9   11   13</code></pre>
<pre class="r"><code>ad$layers[&quot;unspliced&quot;][,c(&quot;var2&quot;, &quot;var3&quot;)]</code></pre>
<pre><code>##    var2 var3
## s1   10   12
## s2   11   13</code></pre>
</div>
<div id="note-on-state" class="section level3">
<h3>Note on state</h3>
<p>If you assign an AnnData object to another variable and
modify either, both will be modified:</p>
<pre class="r"><code>ad2 &lt;- ad

ad$X[,2] &lt;- 10

list(ad = ad$X, ad2 = ad2$X)</code></pre>
<pre><code>## $ad
##        var1 var2     var3
## s1 1.166667   10 5.833333
## s2 1.750000   10 5.250000
## 
## $ad2
##        var1 var2     var3
## s1 1.166667   10 5.833333
## s2 1.750000   10 5.250000</code></pre>
<p>This is standard Python behaviour but not R. In order to
have two separate copies of an AnnData object, use the <code>$copy()</code> function:</p>
<pre class="r"><code>ad3 &lt;- ad$copy()

ad$X[,2] &lt;- c(3, 4)

list(ad = ad$X, ad3 = ad3$X)</code></pre>
<pre><code>## $ad
##        var1 var2     var3
## s1 1.166667    3 5.833333
## s2 1.750000    4 5.250000
## 
## $ad3
##        var1 var2     var3
## s1 1.166667   10 5.833333
## s2 1.750000   10 5.250000</code></pre>
</div>
</div>
<div id="future-work" class="section level2">
<h2>Future work</h2>
<p>In some cases, this package may still act more like a Python package rather than an R package.
Some more helper functions and helper classes need to be defined in order to fully encapsulate
<code>AnnData()</code> objects. Examples are:</p>
<pre class="r"><code>ad$chunked_X(1)</code></pre>
<pre><code>## &lt;generator object AnnData.chunked_X at 0x7f9040bf4258&gt;</code></pre>
<p>Following functionality has not been tested:</p>
<pre class="r"><code>ad$rename_categories(...)
ad$strings_to_categoricals(...)</code></pre>
<p>Currently not implemented are the <code>read_zarr()</code> and <code>ad$write_zarr()</code> functions.
I need some example data to test this functionality.</p>
</div>
<div id="references" class="section level2 unnumbered">
<h2>References</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-wolf_scanpylargescalesinglecell_2018" class="csl-entry">
Wolf, F Alexander, Philipp Angerer, and Fabian J Theis. 2018. <span>“SCANPY: Large-Scale Single-Cell Gene Expression Data Analysis.”</span> <em>Genome Biology</em> 19 (February): 15. <a href="https://doi.org/10.1186/s13059-017-1382-0">https://doi.org/10.1186/s13059-017-1382-0</a>.
</div>
</div>
</div>
