---
abstract: Recent technological breakthroughs in single-cell RNA sequencing are revolutionizing modern experimental design in biology. The increasing size of the single-cell expression data from which networks can be inferred allows identifying more complex, non-linear dependencies between genes. Moreover, the inter-cellular variability that is observed in single-cell expression data can be used to infer not only one global network representing all the cells, but also numerous regulatory networks that are more specific to certain conditions. By experimentally perturbing certain genes, the deconvolution of the true contribution of these genes can also be greatly facilitated. In this chapter, we will therefore tackle the advantages of single-cell transcriptomic data and show how new methods exploit this novel data type to enhance the inference of gene regulatory networks.
authors:
- Helena Todorov
- rcannood
- Wouter Saelens
- Yvan Saeys
date: "2018-12-14T00:00:00Z"
doi: "10.1007/978-1-4939-8882-2_10"
featured: false
image:
  caption: 'Performing network inference on single-cell profiles could result in more precise predicted regulatory interactions.'
  focal_point: ""
  preview_only: false
projects: []
publication: 'Methods in Molecular Biology'
publication_short: ""
publication_types:
- "2"
publishDate: "2018-12-14T00:00:00Z"
# slides: example
summary: A discussion on the advantages of performing network inference with single-cell transcriptomics data.
# tags:
# - Source Themes
title: Network Inference from Single-Cell Transcriptomic Data
# url_code: "https://github.com/dynverse/dynbenchmark"
# url_dataset: "https://doi.org/10.5281/zenodo.1443566"
# url_pdf: https://genomebiology.biomedcentral.com/track/pdf/10.1186/s13059-019-1738-8
# url_poster: ""
# url_project: ""
# url_slides: ""
# url_source: ""
# url_video: ""
---
