---
abstract: 'Trajectory inference has radically enhanced single-cell RNA-seq research by enabling the study of dynamic changes in gene expression. Downstream of trajectory inference, it is vital to discover genes that are (i) associated with the lineages in the trajectory, or (ii) differentially expressed between lineages, to illuminate the underlying biological processes. Current data analysis procedures, however, either fail to exploit the continuous resolution provided by trajectory inference, or fail to pinpoint the exact types of differential expression. We introduce tradeSeq, a powerful generalized additive model framework based on the negative binomial distribution that allows flexible inference of both within-lineage and between-lineage differential expression. By incorporating observation-level weights, the model additionally allows to account for zero inflation. We evaluate the method on simulated datasets and on real datasets from droplet-based and full-length protocols, and show that it yields biological insights through a clear interpretation of the data.'
authors:
- Koen Van den Berge
- Hector Roux de BÃ©zieux
- Kelly Street
- Wouter Saelens
- rcannood
- Yvan Saeys
- Sandrine Dudoit
- Lieven Clement
date: "2020-03-05T00:00:00Z"
doi: 10.1038/s41467-020-14766-3
featured: false
image:
  caption: 'Overview of tradeSeq functionality.'
  focal_point: ''
  preview_only: false
project: []
publication: 'Nature Communications'
publication_short: ''
publication_types:
- '2'
publishDate: '2020-03-05T00:00:00Z'
# slides: example
summary: Discover genes associated with dynamic biological processes.
# tags:
# - Example Tag
title: 'Flexible inference of both within-lineage and between-lineage differential expression.'
url_code: 'https://github.com/statOmics/tradeSeq'
# url_dataset: ''
# url_pdf: ''
# url_poster: ''
# url_slides: ''
# url_source: ''
# url_video: ''
---
