---
abstract: Recent developments in single‐cell transcriptomics have opened new opportunities for studying dynamic processes in immunology in a high throughput and unbiased manner. Starting from a mixture of cells in different stages of a developmental process, unsupervised trajectory inference algorithms aim to automatically reconstruct the underlying developmental path that cells are following. In this review, we break down the strategies used by this novel class of methods, and organize their components into a common framework, highlighting several practical advantages and disadvantages of the individual methods. We also give an overview of new insights these methods have already provided regarding the wiring and gene regulation of cell differentiation. As the trajectory inference field is still in its infancy, we propose several future developments that will ultimately lead to a global and data‐driven way of studying immune cell differentiation.
authors:
- rcannood
- Wouter Saelens
- Yvan Saeys
author_notes:
- Equal contribution
- Equal contribution
date: "2016-09-28T00:00:00Z"
doi: 10.1002/eji.201646347
featured: false
image:
  caption: 'Applications of single‐cell trajectory inference methods.'
  focal_point: ''
  preview_only: false
project: []
publication: 'European Journal of Immunology'
publication_short: ''
publication_types:
- '2'
publishDate: '2016-09-28T00:00:00Z'
# slides: example
summary: We highlight several practical advantages and disadvantages of the different trajectory inference methods.
# tags:
# - Example Tag
title: 'Computational methods for trajectory inference from single-cell transcriptomics'
# url_code: ''
# url_dataset: ''
url_pdf: 'https://onlinelibrary.wiley.com/doi/epdf/10.1002/eji.201646347'
# url_poster: ''
# url_slides: ''
# url_source: ''
# url_video: ''
---
