---
abstract: A critical step in the analysis of large genome-wide gene expression datasets is the use of module detection methods to group genes into co-expression modules. Because of limitations of classical clustering methods, numerous alternative module detection methods have been proposed, which improve upon clustering by handling co-expression in only a subset of samples, modelling the regulatory network, and/or allowing overlap between modules. In this study we use known regulatory networks to do a comprehensive and robust evaluation of these different methods. Overall, decomposition methods outperform all other strategies, while we do not find a clear advantage of biclustering and network inference-based approaches on large gene expression datasets. Using our evaluation workflow, we also investigate several practical aspects of module detection, such as parameter estimation and the use of alternative similarity measures, and conclude with recommendations for the further development of these methods.
authors:
- Wouter Saelens
- <b>Robrecht Cannoodt</b>
- Yvan Saeys
date: "2018-03-15T00:00:00Z"
doi: "10.1038/s41467-018-03424-4"
featured: false
image:
  caption: 'Different types of module detection methods included in the evaluation.'
  focal_point: ""
  preview_only: false
projects: []
publication: 'Nature Communications'
publication_short: ""
publication_types:
- "2"
publishDate: "2018-03-15T00:00:00Z"
# slides: example
summary: A comprehensive and robust evaluation of module detection methods, which group genes into co-expression modules.
# tags:
# - Source Themes
title: A comprehensive evaluation of module detection methods for gene expression data
url_code: "http://www.github.com/saeyslab/moduledetection-evaluation"
# url_dataset: ""
url_pdf: https://www.nature.com/articles/s41467-018-03424-4.pdf
# url_poster: ""
# url_project: ""
# url_slides: ""
# url_source: ""
# url_video: ""
---
