---
abstract: Recent advances in RNA sequencing enable the generation of genome-wide expression data at the single-cell level, opening up new avenues for transcriptomics and systems biology. A new application of single-cell whole-transcriptomics is the unbiased ordering of cells according to their progression along a dynamic process of interest. We introduce SCORPIUS, a method which can effectively reconstruct an ordering of individual cells without any prior information about the dynamic process. Comprehensive evaluation using ten scRNA-seq datasets shows that SCORPIUS consistently outperforms state-of-the-art techniques. We used SCORPIUS to generate novel hypotheses regarding dendritic cell development, which were subsequently validated in vivo. This work enables data-driven investigation and characterization of dynamic processes and lays the foundation for objective benchmarking of future trajectory inference methods.
authors:
- <b>Robrecht Cannoodt</b>
- Wouter Saelens
- Dorine Sichien
- Simon Tavernier
- Sophie Janssens
- Martin Guilliams
- Bart Lambrecht
- Katleen De Preter
- Yvan Saeys
date: "2016-10-07T00:00:00Z"
doi: 10.1101/079509
featured: false
image:
  caption: 'SCORPIUS infers a trajectory in three steps.'
  focal_point: ''
  preview_only: false
project: []
publication: 'bioRxiv'
publication_short: ''
publication_types:
- '3'
publishDate: '2016-10-07T00:00:00Z'
# slides: example
summary: SCORPIUS improves trajectory inference and identifies novel modules in dendritic cell development
# tags:
# - Example Tag
title: 'SCORPIUS improves trajectory inference and identifies novel modules in dendritic cell development'
url_code: 'https://github.com/rcannood/SCORPIUS'
# url_dataset: ''
url_pdf: 'https://www.biorxiv.org/content/biorxiv/early/2016/10/07/079509.full-text.pdf'
# url_poster: ''
# url_slides: ''
# url_source: ''
# url_video: ''
---
